### ISSUE TITLE
Once mods are added the server doesn't start listening

### ISSUE THREAD

**Issue by MarvinKlemp** [FezVrasta/ark-server-tools#1077]
It's a completely fresh install. 
I actually switched from a manual install to arkservertools as I couldn't get any mods running. Maybe this is a general issue ATM ?

I could also add a complete history output with important commands.
That would also help ppl. to get a complete installation instruction for new environments (Ubuntu 18.04).

I used the fix for Linux Servers described in #1063

Starting the server without any addons doen't yield an Issue.

But adding an addon to `/etc/arkmanager/instances/main.cfg ` breaks the server.
I've tested different mods for example adding only 1364327869 or 1786522940.

```
Running command 'status' for instance 'main'
 Server running:   Yes
 Server PID:   5891
 Server listening:   No
 Server build ID:   4776339
```

`cat ShooterGame-b...`
```
﻿[2020.03.16-10.37.49:789][  0]Log file open, 03/16/20 11:37:49
[2020.03.16-10.37.49:789][  0]ARK Version: 307.4
[2020.03.16-10.37.49:789][  0]PID: 5891
[2020.03.16-10.38.22:514][  0]Primal Game Data Took 31.69 seconds
```

`cat /var/log/arktools/arkmanager.log`:
```
2020-03-16 11:50:09: [main] The server is starting...
2020-03-16 11:50:09: [main] The server is now running, and should be up within 10 minutes
```

`cat /var/log/arktools/arkserver.log`
```
2020-03-16 11:50:09: [main] 2020-03-16 11:50:09: start
2020-03-16 11:50:09: [main] 2020-03-16 11:50:09: Running /home/steam/ARK/ShooterGame/Binaries/Linux/ShooterGameServer Valguero_P\?RCONEnabled=True\?RCONPort=32330\?SessionName=HIDDENNAME\ YOLO\?Port=7777\?QueryPort=27015\?ServerPassword=HIDDENPW\?ServerAdminPassword=keyboardcat\?MaxPlayers=20\?GameModIds=1786522940\?listen -log -log
2020-03-16 11:50:09: [main] 2020-03-16 11:50:09: Server PID: 12700
2020-03-16 11:50:10: [main] [S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
2020-03-16 11:50:11: [main] Setting breakpad minidump AppID = 346110
```

**Comment by MarvinKlemp** [comment]
Additional information (whole setup on a fresh server) executed as root:

Note after writing:
Looks like this installation is working on a fresh server. So you can use this to setup on ubuntu 18. 

```
# system requirements
$ apt-get update
$ apt-get upgrade
$ apt-get install perl-modules curl lsof libc6-i386 lib32gcc1 bzip2 libcompress-raw-zlib-perl

# steamcmd
$ add-apt-repository multiverse
$ dpkg --add-architecture i386
$ add-get install steamcmd

# firewall
$ iptables -I INPUT -p udp --dport 27016 -j ACCEPT
$ iptables -I INPUT -p tcp --dport 27016 -j ACCEPT
$ iptables -I INPUT -p udp --dport 7778 -j ACCEPT
$ iptables -I INPUT -p tcp --dport 7778 -j ACCEPT
$ iptables -I INPUT -p tcp --dport 32330 -j ACCEPT

# ark server tools
$ curl -sL http://git.io/vtf5N | sudo bash -s steam
$ arkmanager --version // doesnt find the executables as the default ubuntu config is in /usr/games
$ nano /etc/arkmanager/arkmanager.cfg  // changed executable to /usr/games 
$ arkmanager --version // should work now
$ nano /etc/arkmanager/instances/main.cfg // changed map to Valguero_P, name, password

# installing the server
$ arkmanager --verbose install
>.... (a lot of steam output)
>Update state (0x101) committing, progress: 88.75 (10233253291 / 11530167447)
>Success! App '376030' fully installed.
>CWorkThreadPool::~CWorkThreadPool: work processing queue not empty: 2 items discarded.

# starting the server
$ arkmanager start
$ arkmanager status
>Running command 'status' for instance 'main'
>Server running:   Yes
>Server PID:   16435
>Server listening:   Yes
>Server Name: SomeServerName - (v307.4)
>Players: 0 / 20
>Active Players: 0
>Server online:   Yes
>ARKServers link:   http://arkservers.net/server/49.12.15.70:27015
>Steam connect link:   steam://connect/49.12.15.70:27015
>Server build ID:   4776339

$ arkmanager stop
```

Now if I add an mod. In this case Awesome Spyglass. Is one of the most subscribed so it should work.
https://steamcommunity.com/sharedfiles/filedetails/?id=1404697612


```
# install mod
$ arkmanager installmod 1404697612
>Running command 'installmod' for instance 'main'
>Downloading mod 1404697612 ... Mod 1404697612 downloaded
>Copying files to /home/steam/ARK/ShooterGame/Content/Mods/1404697612
>Mod 1404697612 installed

$ ls -al /home/steam/ARK/ShooterGame/Content/Mods/1404697612
>drwxr-xr-x  3 steam steam  4096 Mar 16 14:18 111111111
>-rwxr-xr-x  1 steam steam   273 Mar 16 14:12 111111111.mod
>drwxr-xr-x  5 steam steam  4096 Mar 16 14:24 1404697612
>-rw-r--r--  1 steam steam   310 Feb 14  2019 1404697612.mod
>drwxr-xr-x  4 steam steam 20480 Mar 16 14:18 Ragnarok
>drwxr-xr-x  5 steam steam  4096 Mar 16 14:18 TheCenter
>drwxr-xr-x  4 steam steam  4096 Mar 16 14:18 Valguero

# enable the mod
$ arkmanager enablemod 1404697612
>Running command 'enablemod' for instance 'main'
>/usr/local/bin/arkmanager: line 2813: /etc/arkmanager/instances/main.cfg: Permission denied
# this is quite interesting and may be an issue.
# I just used nano to chenge it and add 1404697612 to gameModIDS
```

Now comes the server starting part

```
# restart server
$ arkmanager start

$ cat /var/log/arktools/arkserver.log 
>Running /home/steam/ARK/ShooterGame/Binaries/Linux/ShooterGameServer Valguero_P\?>RCONEnabled=True\?RCONPort=32330\?SessionName=NAME\?Port=7777\?>QueryPort=27015\?ServerPassword=PASSWORD\?ServerAdminPassword=keyboardcat\?MaxPlayers=20\?>GameModIds=1404697612\?listen
>2020-03-16 14:32:29: [main] 2020-03-16 14:32:29: Server PID: 24535
>2020-03-16 14:32:30: [main] [S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
>2020-03-16 14:32:30: [main] Setting breakpad minidump AppID = 346110
>2020-03-16 14:34:15: [main] 2020-03-16 14:34:15: server is up

$ arkmanager status
>Running command 'status' for instance 'main'
>Server running:   Yes
>Server PID:   24535
>Server listening:   Yes
>Server Name: SomeServerName - (v307.4)
>Players: 0 / 20
>Active Players: 0
>Server online:   Yes
>ARKServers link:   http://arkservers.net/server/49.12.15.70:27015
>Steam connect link:   steam://connect/49.12.15.70:27015
>Server build ID:   4776339
```

Uff .. looks like it's working now ? As you can see the Mod is In GameModIds when starting + server is listening

Just a quick check on another mod. I also installed the Valguero extension (1786522940) like the Awesome Spyglass above.
https://steamcommunity.com/sharedfiles/filedetails/?id=1786522940

```
$ arkmanager start
>Running command 'status' for instance 'main'
>Server running:   Yes
>Server PID:   29879
>Server listening:   Yes
>Server Name: SomeServerName - (v307.4)
>Players: 0 / 20
>Active Players: 0
>Server online:   Yes
>ARKServers link:   http://arkservers.net/server/49.12.15.70:27015
>Steam connect link:   steam://connect/49.12.15.70:27015
>Server build ID:   4776339
```

**Comment by klightspeed** [comment]
> I used the fix for Linux Servers described in #1063

The fix in #1063 was only required for a short time after Genesis dropped, as when Genesis dropped Wildcard forgot to include the Linux libraries when they pushed out the update.

> ```
> $ arkmanager enablemod 1404697612
> >Running command 'enablemod' for instance 'main'
> >/usr/local/bin/arkmanager: line 2813: /etc/arkmanager/instances/main.cfg: Permission denied
> # this is quite interesting and may be an issue.
> ```

`enablemod` and `disablemod` require that the instance config be writable (they append or replace the `arkmod_{modid}=game` option in the instance config).

**Comment by MarvinKlemp** [comment]
> The fix in #1063 was only required for a short time after Genesis dropped, as when Genesis dropped Wildcard forgot to include the Linux libraries when they pushed out the update.

That could be why the second try setting up from scratch worked as I didn't include the fix anymore.
